gravitee-apim-rest-api/gravitee-apim-rest-api-rest/src/main/java/io/gravitee/rest/api/rest/filter/PermissionsFilter.java

```diff
@@ -91,6 +92,8 @@ private boolean hasPermission(Permission permission, ContainerRequestContext req
                 return hasPermission(executionContext, permission, getApiId(requestContext));
             case GROUP:
                 return hasPermission(executionContext, permission, getGroupId(requestContext));
+            case INTEGRATION:
+                return hasPermission(executionContext, permission, getId(INTEGRATION_ID_PARAM, requestContext));
```

Comment:
We can use switch expression

```java
return switch() {
  case ORGANISATION -> hasPermission(executionContext, permission, executionContext.getOrganizationId());
  // ...
};
```

and replace default by the last one none managed `PLATEFORM`

Reply:
👍